### PR TITLE
Circle cli

### PR DESCRIPTION
**Pull request by AvitalFineRedis**

### CODE REVIEW

File: validators/__init__.py [hunk 1]
```diff
@@ -1,12 +1,11 @@
 import abc
 
 
-VALIDATORS = ['yaml',]
+VALIDATORS = ['yaml', 'cli', ]
 
 
-class BaseValidator(object):
+class BaseValidator(metaclass=abc.ABCMeta):
```
Comment (chayim) [inline validators/__init__.py:7]: This class needs a docstring (definition)

File: validators/__init__.py [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
 
-    @abc.abstractmethod
     def is_valid(self, content):
```
Comment (chayim) [inline validators/__init__.py:10]: Add a docstring for the function

File: validators/cli_validator.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from validators import BaseValidator
+import tempfile
+import os
+
+class CliValidator(BaseValidator):
```
Comment (chayim) [inline validators/cli_validator.py]: Docstring

File: validators/cli_validator.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    
+    def is_valid(self, content:str) -> bool:
```
Comment (chayim) [inline validators/cli_validator.py]: Docstring

File: validators/cli_validator.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        
+        fd, abspath = tempfile.mkstemp(suffix=".yaml")
+        with open(abspath, 'w+') as f:
+            f.write(content)
+
+        try:
+            ex = os.system('circleci config validate %s' % abspath)
```
Comment (chayim) [inline validators/cli_validator.py]: os.system is good, but the new preferred way to do this in python is with the subprocess library. Take a look at this below, and try out subprocess.run. It's a good learning experience.

https://docs.python.org/3/library/subprocess.html

**Review by chayim — APPROVED** [review]
Ready to merge - but take a look at the subprocess.run comment.

File: validators/cli_validator.py (hunk 5 above)
Comment (AvitalFineRedis) [reply validators/cli_validator.py]: I changed it, and it shows me the same error as before, so I hope it works ;)

**Pull request merged**